deltachat-ios/View/Cell/GalleryCell.swift

```diff
@@ -42,32 +50,13 @@ class GalleryCell: UICollectionViewCell {
         playButtonView.constraint(equalTo: CGSize(width: 50, height: 50))
     }
 
-    func update(msg: DcMsg) {
-        guard let viewtype = msg.viewtype, let fileUrl = msg.fileURL else {
-            return
-        }
-        switch viewtype {
-        case .image:
-            imageView.image = msg.image
-            playButtonView.isHidden = true
-        case .video:
-            let key = fileUrl.absoluteString
-            if let image = ThumbnailCache.shared.restoreImage(key: key) {
-                imageView.image = image
-            } else {
-                imageView.loadVideoThumbnail(from: fileUrl, placeholderImage: nil) { thumbnail in
-                    if let image = thumbnail {
-                        ThumbnailCache.shared.storeImage(image: image, key: key)
-                    }
-                }
-            }
-            playButtonView.isHidden = false
-        case .gif:
-            imageView.sd_setImage(with: fileUrl, placeholderImage: nil)
```

Comment:
instead of working on the imageView we should try to load the gif with the help of `SDAnimatedImage(data: gifData)`